### PR TITLE
Tower 3.2.2 vault credential type

### PR DESCRIPTION
**Pull request by jameswnl**
Added `vault` type

screen shot 2018-02-14 at 4 46 12 pm

### CODE REVIEW

**Comment by jameswnl** [comment]
@gmcculloug @syncrou green now

**Comment by jameswnl** [comment]
@miq-bot remove_label wip

**Comment by jameswnl** [comment]
@miq-bot add_label enhancement

File: spec/support/ansible_shared/automation_manager/refresher.rb [hunk 1]
```diff
@@ -113,6 +113,11 @@ def assert_counts
   def assert_credentials
     expect(expected_configuration_script.authentications.count).to eq(3)
 
+    # vault_credential
+    vault_credential = Authentication.all.find_by(:type => manager_class::VaultCredential)
+    expect(vault_credential.options.keys).to match_array([:vault_password])
+    expect(vault_credential.name).to eq("Demo Creds 2")
```
Comment (syncrou) [inline spec/support/ansible_shared/automation_manager/refresher.rb]: @jameswnl - I'm not sure I'm seeing a reason to assert this name as it doesn't help determine the `vault_credential` Type. 

Do you think there would be value in asserting that the vault credential is really a `machine_credential` that includes a kind of `ssh` and a non empty `vault_password`?

File: spec/support/ansible_shared/automation_manager/refresher.rb [hunk 2]
```diff
@@ -121,8 +126,7 @@ def assert_credentials
       :name   => "hello_machine_cred",
       :userid => "admin",
     )
-    expect(machine_credential.options.keys).to match_array([:become_method, :become_password, :become_username, :ssh_key_data, :ssh_key_unlock, :vault_password])
-    expect(machine_credential.options.keys).to match_array(machine_credential.class::EXTRA_ATTRIBUTES.keys)
```
Comment (syncrou) [inline spec/support/ansible_shared/automation_manager/refresher.rb]: So this isn't assertable anymore for `machine_credential`?

Comment (jameswnl) [reply spec/support/ansible_shared/automation_manager/refresher.rb]: It is still assertable. L125 was an unintended duplicate and so removed this time. L129 is the modified L124

File: spec/support/ansible_shared/automation_manager/refresher.rb (hunk 1 above)
Comment (jameswnl) [reply spec/support/ansible_shared/automation_manager/refresher.rb]: There's really nothing else to check as `vault_credential` really has only 1 piece of info interesting,  i.e. the `vault_password` and it is inside the hash stored in the `options` column.

In the parsing, we already make a decision to classify this as a `vault_credential` and not related to `machine_credential`.

I've added a check to make sure the `options.vault_password` not empty.

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit a63f9fb167b9d5c542857ced73b08a1dd71f600e
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Wed Feb 21 13:12:34 2018 -0500

    Merge pull request #54 from jameswnl/vault
    
    Tower 3.2.2 vault credential type
    (cherry picked from commit 946a58e3fc14def1bd6f595739245d855b468efe)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1559479
```

**Pull request merged**